### PR TITLE
refactor(api-client,core): Migrate tests to TypeScript

### PR DESCRIPTION
**Pull request by bennyn**

### CODE REVIEW

File: packages/api-client/src/main/conversation/AccessUpdate.ts [hunk 1]
```diff
@@ -0,0 +1,25 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
```
Comment (ffflorian) [inline packages/api-client/src/main/conversation/AccessUpdate.ts]:
```suggestion
 * Copyright (C) 2019 Wire Swiss GmbH
```

File: packages/api-client/src/main/conversation/CodeUpdate.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
```
Comment (ffflorian) [inline packages/api-client/src/main/conversation/CodeUpdate.ts]:
```suggestion
 * Copyright (C) 2019 Wire Swiss GmbH
```

File: packages/api-client/src/main/conversation/ConnectRequest.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
```
Comment (ffflorian) [inline packages/api-client/src/main/conversation/ConnectRequest.ts]:
```suggestion
 * Copyright (C) 2019 Wire Swiss GmbH
```

File: packages/api-client/src/main/conversation/MemberJoin.ts [hunk 4]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
```
Comment (ffflorian) [inline packages/api-client/src/main/conversation/MemberJoin.ts]:
```suggestion
 * Copyright (C) 2019 Wire Swiss GmbH
```

File: packages/api-client/src/main/conversation/MemberLeave.ts [hunk 5]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
```
Comment (ffflorian) [inline packages/api-client/src/main/conversation/MemberLeave.ts]:
```suggestion
 * Copyright (C) 2019 Wire Swiss GmbH
```

File: packages/api-client/src/main/conversation/OtrMessageAdd.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
```
Comment (ffflorian) [inline packages/api-client/src/main/conversation/OtrMessageAdd.ts]:
```suggestion
 * Copyright (C) 2019 Wire Swiss GmbH
```

File: packages/api-client/src/main/conversation/Rename.ts [hunk 7]
```diff
@@ -0,0 +1,22 @@
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
```
Comment (ffflorian) [inline packages/api-client/src/main/conversation/Rename.ts]:
```suggestion
 * Copyright (C) 2019 Wire Swiss GmbH
```

File: packages/core/src/main/Account.test.node.ts [hunk 8]
```diff
@@ -17,20 +17,37 @@
...
+} from '@wireapp/api-client/dist/commonjs/event';
+import {BackendErrorLabel, StatusCode} from '@wireapp/api-client/dist/commonjs/http';
+import {NotificationAPI} from '@wireapp/api-client/dist/commonjs/notification';
+import {WebSocketTopic} from '@wireapp/api-client/dist/commonjs/tcp';
+import {ValidationUtil} from '@wireapp/commons';
+import {GenericMessage, Text} from '@wireapp/protocol-messaging';
+import {MemoryEngine} from '@wireapp/store-engine';
+import {Server as MockSocketServer} from 'mock-socket';
+import nock = require('nock');
+import {Account} from './Account';
```
Comment (ffflorian) [inline packages/core/src/main/Account.test.node.ts]:
```suggestion

import {Account} from './Account';
```

File: packages/core/src/main/auth/LoginSanitizer.test.node.ts [hunk 9]
```diff
@@ -16,31 +16,34 @@
  * along with this program. If not, see http://www.gnu.org/licenses/.
  *
  */
-
-const auth = require('../../../dist/auth/');
+import {LoginData} from '@wireapp/api-client/dist/commonjs/auth/';
+import {ClientType} from '@wireapp/api-client/dist/commonjs/client';
+import {LoginSanitizer} from './LoginSanitizer';
```
Comment (ffflorian) [inline packages/core/src/main/auth/LoginSanitizer.test.node.ts]:
```suggestion

import {LoginSanitizer} from './LoginSanitizer';
```

File: packages/core/src/main/conversation/MessageTimer.test.node.ts [hunk 10]
```diff
@@ -1,4 +1,22 @@
-const {MessageTimer} = require('../../../dist/conversation/');
+/*
+ * Wire
+ * Copyright (C) 2018 Wire Swiss GmbH
```
Comment (ffflorian) [inline packages/core/src/main/conversation/MessageTimer.test.node.ts]:
```suggestion
 * Copyright (C) 2019 Wire Swiss GmbH
```

File: packages/core/src/main/cryptography/CryptographyService.test.node.ts [hunk 11]
```diff
@@ -16,32 +16,30 @@
...
-
-async function createEngine(storeName) {
+import {APIClient} from '@wireapp/api-client';
+import {Cryptobox} from '@wireapp/cryptobox';
+import * as Proteus from '@wireapp/proteus';
+import {CRUDEngine, MemoryEngine} from '@wireapp/store-engine';
+import * as bazinga64 from 'bazinga64';
+import * as crypto from 'crypto';
+import {promisify} from 'util';
+import * as CryptographyHelper from '../test/CryptographyHelper';
```
Comment (ffflorian) [inline packages/core/src/main/cryptography/CryptographyService.test.node.ts]:
```suggestion

import * as CryptographyHelper from '../test/CryptographyHelper';
```

File: packages/core/src/main/test/CryptographyHelper.ts [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+
+import {Cryptobox} from '@wireapp/cryptobox';
+import * as Proteus from '@wireapp/proteus';
+import {PreKey} from '@wireapp/proteus/dist/keys';
+import * as bazinga64 from 'bazinga64';
+import {CryptographyService} from '../cryptography';
```
Comment (ffflorian) [inline packages/core/src/main/test/CryptographyHelper.ts]:
```suggestion

import {CryptographyService} from '../cryptography';
```

File: packages/api-client/src/main/conversation/AccessUpdate.ts (hunk 1 above)
Comment (Yserz) [inline packages/api-client/src/main/conversation/AccessUpdate.ts]:
```suggestion
 * Copyright (C) 2019 Wire Swiss GmbH
```

File: packages/core/src/main/test/CryptographyHelper.ts [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+
+const StoreHelper = require('./StoreHelper');
```
Comment (ffflorian) [inline packages/core/src/main/test/CryptographyHelper.ts]: Why not rewrite the StoreHelper, too?

File: packages/core/src/main/test/PayloadHelper.ts [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ *
+ */
+const faker = require('faker');
```
Comment (ffflorian) [inline packages/core/src/main/test/PayloadHelper.ts]:
```suggestion

const faker = require('faker');
```

File: packages/api-client/src/main/conversation/CodeUpdate.ts (hunk 2 above)
Comment (Yserz) [inline packages/api-client/src/main/conversation/CodeUpdate.ts]:
```suggestion
 * Copyright (C) 2019 Wire Swiss GmbH
```

File: packages/core/src/main/test/PayloadHelper.ts (hunk 14 above)
Comment (ffflorian) [reply packages/core/src/main/test/PayloadHelper.ts]: Also, `faker` has types: https://www.npmjs.com/package/@types/faker

File: packages/core/src/main/Account.test.node.ts [hunk 15]
```diff
@@ -17,20 +17,37 @@
...
+  UserConnectionEvent,
+} from '@wireapp/api-client/dist/commonjs/event';
+import {BackendErrorLabel, StatusCode} from '@wireapp/api-client/dist/commonjs/http';
+import {NotificationAPI} from '@wireapp/api-client/dist/commonjs/notification';
+import {WebSocketTopic} from '@wireapp/api-client/dist/commonjs/tcp';
+import {ValidationUtil} from '@wireapp/commons';
+import {GenericMessage, Text} from '@wireapp/protocol-messaging';
+import {MemoryEngine} from '@wireapp/store-engine';
+import {Server as MockSocketServer} from 'mock-socket';
+import nock = require('nock');
```
Comment (Yserz) [inline packages/core/src/main/Account.test.node.ts]: Can we use a proper import for nock by now?
https://github.com/nock/nock/issues/1670

Comment (ffflorian) [reply packages/core/src/main/Account.test.node.ts]: [Nope](https://github.com/nock/nock/blob/master/types/index.d.ts#L8) (at least not without `esModuleInterop: true`)

Comment (Yserz) [reply packages/core/src/main/Account.test.node.ts]: aww :/ pity

File: packages/core/src/main/test/CryptographyHelper.ts (hunk 13 above)
Comment (bennyn) [reply packages/core/src/main/test/CryptographyHelper.ts]: Because it's currently used in some JS code (browser tests).

**Pull request merged**